specification/guestconfiguration/GuestConfiguration.Management/GuestConfigurationAssignment.tsp

```diff
@@ -0,0 +1,443 @@
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridCompute/machines/{machineName}/providers/Microsoft.GuestConfiguration/guestConfigurationAssignments/{guestConfigurationAssignmentName}")
+  createOrUpdate is GuestConfigurationHCRPAssignmentOps.CreateOrUpdateSync<GuestConfigurationAssignment>;
+
+  /**
+   * Delete a guest configuration assignment
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-delete-operation-response-codes" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @operationId("GuestConfigurationHCRPAssignments_Delete")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridCompute/machines/{machineName}/providers/Microsoft.GuestConfiguration/guestConfigurationAssignments/{guestConfigurationAssignmentName}")
+  delete is GuestConfigurationHCRPAssignmentOps.DeleteSync<
+    GuestConfigurationAssignment,
+    Response = ArmDeletedResponse
+  >;
+
+  /**
+   * List all guest configuration assignments for an ARC machine.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @operationId("GuestConfigurationHCRPAssignments_List")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridCompute/machines/{machineName}/providers/Microsoft.GuestConfiguration/guestConfigurationAssignments")
+  list is GuestConfigurationHCRPAssignmentOps.List<
+    GuestConfigurationAssignment,
+    Response = ArmResponse<GuestConfigurationAssignmentList>
+  >;
+
+  /**
+   * List all reports for the guest configuration assignment, latest report first.
+   */
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @get
+  @action("reports")
+  @operationId("GuestConfigurationHCRPAssignmentReports_List")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridCompute/machines/{machineName}/providers/Microsoft.GuestConfiguration/guestConfigurationAssignments/{guestConfigurationAssignmentName}/reports")
+  guestConfigurationHCRPAssignmentReportsList is GuestConfigurationHCRPAssignmentOps.ActionSync<
+    GuestConfigurationAssignment,
+    void,
+    ArmResponse<GuestConfigurationAssignmentReportList>
+  >;
+
+  /**
+   * Get a report for the guest configuration assignment, by reportId.
+   */
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
```

Comment:
Can the casing-style justification just be that its backcompat / grandfathered in? I don't think we care enough about casing style to justify a breaking change in general.

Reply:
When we later try to remove this suppression, we will discuss the casing stuff case by case.